telemetrylib/library.cpp

```diff
@@ -0,0 +1,49 @@
+#include "library.h"
+
+Telemetry::Telemetry() {}
+Telemetry::Telemetry(std::vector<DTI*> commChannels) {
+    for(int i = 0 ; i < commChannels.size() ; i ++) {
+        this->comm.push_back(commChannels[i]);
+        connect(comm[i], SIGNAL(connectionStatusChanged()), this, SLOT(comChannelChanged()));
+    }
+    qDebug() << "comm channels initialized: "<<comm.size();
+}
+
+void Telemetry::sendData(QByteArray bytes) {
+    qDebug() << "send data current comm channel: "<<commChannel;
+    if(commChannel != -1 && !resendQueue.isBusy()) {
+        comm[commChannel]->sendData(bytes);
+    } else {
+       resendQueue.addToQueue(bytes);
+    }
+}
+
+std::string Telemetry::receiveData() {
+    if (commChannel != -1) {
+        return comm[commChannel]->receiveData();
+    }
+    return "nada";
+}
```

Comment:
This goes for `receiveData()` in `DTI.h`, `SQL.cpp`, and `TCP.cpp` as well:
- In DTI.h/SQL.cpp/TCP.cpp, it looks like this is being implemented as a method for reading a message that was received from whatever the app is connected to. However, it doesn't look like it's ever actually used since the readReply() slot is used for doing this. I think it would be a good idea to convert receiveData() to a method for explicitly requesting data from whatever the pi is connected to (i.e. give the method an argument for the message to send or the type of data to request from the pi's partner, and the pi will send that request for data). Then, readReply() can still be used to read the response from the pi's partner
- Given the above point: I think this should be renamed to something like `getData()` or `requestData()`

Reply:
the tcp class has a functional read reply, but sql does not since the slot cleared all the buffers, but since we are not currently reading anything and implementing readReply in sql class would probably need changes in server script, I think we should put aside this task for now.

Reply:
I agree, we could leave implementing a getData() function as a separate addition for another task, in which case, I would either delete the receiveData() function or at the very least comment it out, as it isn't utilized anywhere (except in TCP.cpp, but we aren't reading any messages from the engineering dashboard).
Also, SQL has a functional readReply(). It's used to get the table name from the server. However, TCP doesn't have readReply() implemented, though it could be implemented as a slot. I don't see any case where we would explicitly call receiveData() since it's a public function instead of connecting the readReply() slot to read a message as soon as it receives one from the partner host.

Reply:
No more action needed, but I'm keeping the thread unresolved for future clarity/reference